### ISSUE TITLE
test_client_no_intermediate test failure

### ISSUE THREAD

**Issue by mgorny** [urllib3/urllib3#1766]
Splitting as requested. I'm going to include the IPv6 as well because I'm not really sure which parts of the reports apply to which tests. With flaky added, this output gets really confusing.

```
==================================================================== FAILURES =====================================================================
Exception in thread Thread-120:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/tmp/urllib3/dummyserver/server.py", line 145, in run
    self.server = self._start_server()
  File "/tmp/urllib3/dummyserver/server.py", line 141, in _start_server
    self.socket_handler(sock)
  File "/tmp/urllib3/test/with_dummyserver/test_socketlevel.py", line 1331, in socket_handler
    ca_certs=DEFAULT_CA,
  File "/usr/lib64/python2.7/ssl.py", line 931, in wrap_socket
    ciphers=ciphers)
  File "/usr/lib64/python2.7/ssl.py", line 599, in __init__
    self.do_handshake()
  File "/usr/lib64/python2.7/ssl.py", line 828, in do_handshake
    self._sslobj.do_handshake()
error: [Errno 0] Error

Exception in thread Thread-119:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/tmp/urllib3/dummyserver/server.py", line 145, in run
    self.server = self._start_server()
  File "/tmp/urllib3/dummyserver/server.py", line 141, in _start_server
    self.socket_handler(sock)
  File "/tmp/urllib3/test/with_dummyserver/test_socketlevel.py", line 1331, in socket_handler
    ca_certs=DEFAULT_CA,
  File "/usr/lib64/python2.7/ssl.py", line 931, in wrap_socket
    ciphers=ciphers)
  File "/usr/lib64/python2.7/ssl.py", line 599, in __init__
    self.do_handshake()
  File "/usr/lib64/python2.7/ssl.py", line 828, in do_handshake
    self._sslobj.do_handshake()
error: [Errno 0] Error

Exception in thread Thread-121:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/tmp/urllib3/dummyserver/server.py", line 145, in run
    self.server = self._start_server()
  File "/tmp/urllib3/dummyserver/server.py", line 141, in _start_server
    self.socket_handler(sock)
  File "/tmp/urllib3/test/with_dummyserver/test_socketlevel.py", line 1331, in socket_handler
    ca_certs=DEFAULT_CA,
  File "/usr/lib64/python2.7/ssl.py", line 931, in wrap_socket
    ciphers=ciphers)
  File "/usr/lib64/python2.7/ssl.py", line 599, in __init__
    self.do_handshake()
  File "/usr/lib64/python2.7/ssl.py", line 828, in do_handshake
    self._sslobj.do_handshake()
error: [Errno 0] Error

Exception in thread Thread-122:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/tmp/urllib3/dummyserver/server.py", line 145, in run
    self.server = self._start_server()
  File "/tmp/urllib3/dummyserver/server.py", line 141, in _start_server
    self.socket_handler(sock)
  File "/tmp/urllib3/test/with_dummyserver/test_socketlevel.py", line 1331, in socket_handler
    ca_certs=DEFAULT_CA,
  File "/usr/lib64/python2.7/ssl.py", line 931, in wrap_socket
    ciphers=ciphers)
  File "/usr/lib64/python2.7/ssl.py", line 599, in __init__
    self.do_handshake()
  File "/usr/lib64/python2.7/ssl.py", line 828, in do_handshake
    self._sslobj.do_handshake()
error: [Errno 0] Error

______________________________________________________ TestHTTPS.test_client_no_intermediate ______________________________________________________
Traceback (most recent call last):
  File "/tmp/urllib3/test/with_dummyserver/test_https.py", line 139, in test_client_no_intermediate
    https_pool.request("GET", "/certificate", retries=False)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py", line 76, in request
    method, url, fields=fields, headers=headers, **urlopen_kw
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py", line 97, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 720, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py", line 376, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 672, in urlopen
    chunked=chunked,
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 387, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib64/python2.7/httplib.py", line 1069, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib64/python2.7/httplib.py", line 1109, in _send_request
    self.endheaders(body)
  File "/usr/lib64/python2.7/httplib.py", line 1065, in endheaders
    self._send_output(message_body)
  File "/usr/lib64/python2.7/httplib.py", line 892, in _send_output
    self.send(msg)
  File "/usr/lib64/python2.7/httplib.py", line 868, in send
    self.sock.sendall(data)
  File "/usr/lib64/python2.7/ssl.py", line 741, in sendall
    v = self.send(data[count:])
  File "/usr/lib64/python2.7/ssl.py", line 707, in send
    v = self._sslobj.write(data)
ProtocolError: ('Connection aborted.', error(104, 'Connection reset by peer'))
---------------------------------------------------------------- Captured log call ----------------------------------------------------------------
WARNING  tornado.general:iostream.py:1518 SSL Error on 17 ('127.0.0.1', 36088): [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
---------------------------------------------------------------- Captured log call ----------------------------------------------------------------
WARNING  tornado.general:iostream.py:1518 SSL Error on 17 ('127.0.0.1', 36092): [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
__________________________________________________ TestHTTPS_TLSv1_3.test_client_no_intermediate __________________________________________________
Traceback (most recent call last):
  File "/tmp/urllib3/test/with_dummyserver/test_https.py", line 139, in test_client_no_intermediate
    https_pool.request("GET", "/certificate", retries=False)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py", line 76, in request
    method, url, fields=fields, headers=headers, **urlopen_kw
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py", line 97, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 720, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py", line 376, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 672, in urlopen
    chunked=chunked,
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 387, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib64/python2.7/httplib.py", line 1069, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib64/python2.7/httplib.py", line 1109, in _send_request
    self.endheaders(body)
  File "/usr/lib64/python2.7/httplib.py", line 1065, in endheaders
    self._send_output(message_body)
  File "/usr/lib64/python2.7/httplib.py", line 892, in _send_output
    self.send(msg)
  File "/usr/lib64/python2.7/httplib.py", line 868, in send
    self.sock.sendall(data)
  File "/usr/lib64/python2.7/ssl.py", line 741, in sendall
    v = self.send(data[count:])
  File "/usr/lib64/python2.7/ssl.py", line 707, in send
    v = self._sslobj.write(data)
ProtocolError: ('Connection aborted.', error(104, 'Connection reset by peer'))
---------------------------------------------------------------- Captured log call ----------------------------------------------------------------
WARNING  tornado.general:iostream.py:1518 SSL Error on 17 ('127.0.0.1', 42400): [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
---------------------------------------------------------------- Captured log call ----------------------------------------------------------------
WARNING  tornado.general:iostream.py:1518 SSL Error on 17 ('127.0.0.1', 42402): [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
_________________________________________________ TestIPv6HTTPProxyManager.test_basic_ipv6_proxy __________________________________________________
Traceback (most recent call last):
  File "/tmp/urllib3/test/with_dummyserver/test_proxy_poolmanager.py", line 363, in test_basic_ipv6_proxy
    r = http.request("GET", "%s/" % self.http_url)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py", line 76, in request
    method, url, fields=fields, headers=headers, **urlopen_kw
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py", line 97, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/poolmanager.py", line 466, in urlopen
    return super(ProxyManager, self).urlopen(method, url, redirect=redirect, **kw)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/poolmanager.py", line 328, in urlopen
    response = conn.urlopen(method, url, **kw)
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 760, in urlopen
    **response_kw
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 760, in urlopen
    **response_kw
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 760, in urlopen
    **response_kw
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py", line 720, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py", line 436, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
MaxRetryError: HTTPConnectionPool(host='::1', port=39839): Max retries exceeded with url: http://localhost:36895/ (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83d48d0>: Failed to establish a new connection: [Errno 111] Connection refused',)))
-------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------
Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e8405750>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e8405750>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cb210>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cb210>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbd90>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbd90>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
---------------------------------------------------------------- Captured log call ----------------------------------------------------------------
WARNING  urllib3.connectionpool:connectionpool.py:745 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e8405750>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
WARNING  urllib3.connectionpool:connectionpool.py:745 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cb210>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
WARNING  urllib3.connectionpool:connectionpool.py:745 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbd90>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:45733/
-------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------
Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbb10>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbb10>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cb1d0>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cb1d0>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbe10>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbe10>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
---------------------------------------------------------------- Captured log call ----------------------------------------------------------------
WARNING  urllib3.connectionpool:connectionpool.py:745 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbb10>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
WARNING  urllib3.connectionpool:connectionpool.py:745 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cb1d0>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
WARNING  urllib3.connectionpool:connectionpool.py:745 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83cbe10>: Failed to establish a new connection: [Errno 111] Connection refused',))': http://localhost:36895/
================================================================ warnings summary =================================================================
test/test_ssl.py::test_context_sni_with_ip_address[False-www.python.org-False]
  /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/ssl_.py:380: SNIMissingWarning: An HTTPS request has been made, but the SNI (Server Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
    SNIMissingWarning,

test/contrib/test_pyopenssl.py::TestHTTPS::test_https_timeout
test/contrib/test_pyopenssl.py::TestHTTPS::test_tunnel
test/contrib/test_pyopenssl.py::TestHTTPS_TLSv1::test_https_timeout
test/contrib/test_pyopenssl.py::TestHTTPS_TLSv1::test_tunnel
test/contrib/test_pyopenssl.py::TestHTTPS_TLSv1_1::test_https_timeout
test/contrib/test_pyopenssl.py::TestHTTPS_TLSv1_1::test_tunnel
test/contrib/test_pyopenssl.py::TestHTTPS_TLSv1_2::test_https_timeout
test/contrib/test_pyopenssl.py::TestHTTPS_TLSv1_2::test_tunnel
test/contrib/test_pyopenssl.py::TestHTTPS_TLSv1_3::test_https_timeout
test/contrib/test_pyopenssl.py::TestHTTPS_TLSv1_3::test_tunnel
test/with_dummyserver/test_https.py::TestHTTPS::test_https_timeout
test/with_dummyserver/test_https.py::TestHTTPS::test_tunnel
test/with_dummyserver/test_https.py::TestHTTPS_TLSv1::test_https_timeout
test/with_dummyserver/test_https.py::TestHTTPS_TLSv1::test_tunnel
test/with_dummyserver/test_https.py::TestHTTPS_TLSv1_1::test_https_timeout
test/with_dummyserver/test_https.py::TestHTTPS_TLSv1_1::test_tunnel
test/with_dummyserver/test_https.py::TestHTTPS_TLSv1_2::test_https_timeout
test/with_dummyserver/test_https.py::TestHTTPS_TLSv1_2::test_tunnel
test/with_dummyserver/test_https.py::TestHTTPS_TLSv1_3::test_https_timeout
test/with_dummyserver/test_https.py::TestHTTPS_TLSv1_3::test_tunnel
test/with_dummyserver/test_no_ssl.py::TestHTTPSWithoutSSL::test_simple
test/with_dummyserver/test_proxy_poolmanager.py::TestHTTPProxyManager::test_proxy_pooling
test/with_dummyserver/test_proxy_poolmanager.py::TestHTTPProxyManager::test_proxy_pooling
test/with_dummyserver/test_proxy_poolmanager.py::TestHTTPProxyManager::test_proxy_pooling
test/with_dummyserver/test_proxy_poolmanager.py::TestHTTPProxyManager::test_proxy_pooling
test/with_dummyserver/test_socketlevel.py::TestProxyManager::test_connect_ipv6_addr
  /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
    InsecureRequestWarning,

test/contrib/test_pyopenssl.py::TestHTTPS_IPv6Addr::test_strip_square_brackets_before_validating
test/with_dummyserver/test_https.py::TestHTTPS_IPv6Addr::test_strip_square_brackets_before_validating
  /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connection.py:384: SubjectAltNameWarning: Certificate for ::1 has no `subjectAltName`, falling back to check for a `commonName` for now. This feature is being removed by major browsers and deprecated by RFC 2818. (See https://github.com/urllib3/urllib3/issues/497 for details.)
    SubjectAltNameWarning,

test/contrib/test_pyopenssl.py::TestSNI::test_hostname_in_first_request_packet
  /tmp/urllib3/dummyserver/server.py:128: NoIPv6Warning: No IPv6 support. Falling back to IPv4.
    warnings.warn("No IPv6 support. Falling back to IPv4.", NoIPv6Warning)

test/with_dummyserver/test_connectionpool.py::TestConnectionPool::test_source_address
  /tmp/urllib3/test/with_dummyserver/test_connectionpool.py:688: NoIPv6Warning: No IPv6 support: skipping.
    warnings.warn("No IPv6 support: skipping.", NoIPv6Warning)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===Flaky Test Report===

test_client_no_intermediate failed (1 runs remaining out of 2).
	<class 'urllib3.exceptions.ProtocolError'>
	('Connection aborted.', error(104, 'Connection reset by peer'))
	[<TracebackEntry /tmp/urllib3/test/with_dummyserver/test_https.py:139>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:76>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:97>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:720>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py:376>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:672>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:387>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1069>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1109>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1065>, <TracebackEntry /usr/lib64/python2.7/httplib.py:892>, <TracebackEntry /usr/lib64/python2.7/httplib.py:868>, <TracebackEntry /usr/lib64/python2.7/ssl.py:741>, <TracebackEntry /usr/lib64/python2.7/ssl.py:707>]
test_client_no_intermediate failed; it passed 0 out of the required 1 times.
	<class 'urllib3.exceptions.ProtocolError'>
	('Connection aborted.', error(104, 'Connection reset by peer'))
	[<TracebackEntry /tmp/urllib3/test/with_dummyserver/test_https.py:139>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:76>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:97>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:720>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py:376>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:672>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:387>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1069>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1109>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1065>, <TracebackEntry /usr/lib64/python2.7/httplib.py:892>, <TracebackEntry /usr/lib64/python2.7/httplib.py:868>, <TracebackEntry /usr/lib64/python2.7/ssl.py:741>, <TracebackEntry /usr/lib64/python2.7/ssl.py:707>]
test_client_no_intermediate failed (1 runs remaining out of 2).
	<class 'urllib3.exceptions.ProtocolError'>
	('Connection aborted.', error(104, 'Connection reset by peer'))
	[<TracebackEntry /tmp/urllib3/test/with_dummyserver/test_https.py:139>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:76>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:97>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:720>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py:376>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:672>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:387>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1069>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1109>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1065>, <TracebackEntry /usr/lib64/python2.7/httplib.py:892>, <TracebackEntry /usr/lib64/python2.7/httplib.py:868>, <TracebackEntry /usr/lib64/python2.7/ssl.py:741>, <TracebackEntry /usr/lib64/python2.7/ssl.py:707>]
test_client_no_intermediate failed; it passed 0 out of the required 1 times.
	<class 'urllib3.exceptions.ProtocolError'>
	('Connection aborted.', error(104, 'Connection reset by peer'))
	[<TracebackEntry /tmp/urllib3/test/with_dummyserver/test_https.py:139>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:76>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:97>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:720>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py:376>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:672>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:387>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1069>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1109>, <TracebackEntry /usr/lib64/python2.7/httplib.py:1065>, <TracebackEntry /usr/lib64/python2.7/httplib.py:892>, <TracebackEntry /usr/lib64/python2.7/httplib.py:868>, <TracebackEntry /usr/lib64/python2.7/ssl.py:741>, <TracebackEntry /usr/lib64/python2.7/ssl.py:707>]
test_basic_ipv6_proxy failed (1 runs remaining out of 2).
	<class 'urllib3.exceptions.MaxRetryError'>
	HTTPConnectionPool(host='::1', port=38641): Max retries exceeded with url: http://localhost:45733/ (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e8493b50>: Failed to establish a new connection: [Errno 111] Connection refused',)))
	[<TracebackEntry /tmp/urllib3/test/with_dummyserver/test_proxy_poolmanager.py:363>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:76>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:97>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/poolmanager.py:466>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/poolmanager.py:328>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:760>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:760>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:760>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:720>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py:436>]
test_basic_ipv6_proxy failed; it passed 0 out of the required 1 times.
	<class 'urllib3.exceptions.MaxRetryError'>
	HTTPConnectionPool(host='::1', port=39839): Max retries exceeded with url: http://localhost:36895/ (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83e83d48d0>: Failed to establish a new connection: [Errno 111] Connection refused',)))
	[<TracebackEntry /tmp/urllib3/test/with_dummyserver/test_proxy_poolmanager.py:363>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:76>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/request.py:97>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/poolmanager.py:466>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/poolmanager.py:328>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:760>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:760>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:760>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/connectionpool.py:720>, <TracebackEntry /tmp/urllib3/.nox/test-2-7/lib/python2.7/site-packages/urllib3/util/retry.py:436>]

===End Flaky Test Report===
============================================================= short test summary info =============================================================
SKIPPED [2] /tmp/urllib3/test/with_dummyserver/test_https.py:672: Skipping base test class
SKIPPED [2] test/test_util.py:461: only run if brotlipy is absent
SKIPPED [1] test/with_dummyserver/test_socketlevel.py:1579: Header parsing errors not available
SKIPPED [1] /tmp/urllib3/test/__init__.py:90: test_parse_url_bytes_type_error_python_3 requires Python3.x to run
SKIPPED [1] /tmp/urllib3/test/__init__.py:90: test_httplib_headers_case_insensitive requires Python3.x to run
SKIPPED [2] /tmp/urllib3/test/with_dummyserver/test_https.py:656: Skipping base test class
SKIPPED [1] test/with_dummyserver/test_socketlevel.py:1585: Header parsing errors not available
SKIPPED [1] test/with_dummyserver/test_socketlevel.py:1582: Header parsing errors not available
SKIPPED [163] /tmp/urllib3/test/contrib/test_securetransport.py:21: Could not import SecureTransport: ImportError('The library Security could not be found',)
FAILED test/with_dummyserver/test_https.py::TestHTTPS::test_client_no_intermediate - ProtocolError: ('Connection aborted.', error(104, 'Connecti...
FAILED test/with_dummyserver/test_https.py::TestHTTPS_TLSv1_3::test_client_no_intermediate - ProtocolError: ('Connection aborted.', error(104, '...
FAILED test/with_dummyserver/test_proxy_poolmanager.py::TestIPv6HTTPProxyManager::test_basic_ipv6_proxy - MaxRetryError: HTTPConnectionPool(host...
======================================== 3 failed, 1140 passed, 174 skipped, 31 warnings in 173.34 seconds ========================================
```

**Comment by hodbn** [comment]
Can you point to a failing CI build or describe your setup?

**Comment by mgorny** [comment]
That would be a good question back in December. I suppose this bug was open as a followup of https://github.com/urllib3/urllib3/issues/1755#issuecomment-557516798. IIRC the test was flaky, i.e. failed or succeeded randomly back then.

It might've been pretty much everything back then, including some race condition. I can't reproduce it right now, and I've recently had a hardware failure meaning I can't test it under the same or even similar conditions (my old hardware was ~15 years old).

I've tried downgrading tornado and openssl to the versions from the time when I reported it. Still no luck. If you have any other ideas, I can try them. Otherwise, I guess it's fine to close this.

I'm going to reenable these tests on Gentoo, and see if anyone else reports them.
